source/contracts/extensions/MarketFeeCalculator.sol

```diff
@@ -28,41 +30,44 @@ contract MarketFeeCalculator {
         uint256 _invalidMarketsInPreviousWindow = _reportingWindow.getNumInvalidMarkets();
         uint256 _previousValidityBondInAttoeth = validityBondInAttoeth[_previousReportingWindow];
 
-        _currentValidityBondInAttoeth = calculateValidityBond(_invalidMarketsInPreviousWindow, _totalMarketsInPreviousWindow, TARGET_INVALID_MARKETS_DIVISOR, _previousValidityBondInAttoeth);
+        _currentValidityBondInAttoeth = calculateFloatingValue(_invalidMarketsInPreviousWindow, _totalMarketsInPreviousWindow, TARGET_INVALID_MARKETS_DIVISOR, _previousValidityBondInAttoeth, Reporting.defaultValidityBond());
         validityBondInAttoeth[_reportingWindow] = _currentValidityBondInAttoeth;
         return _currentValidityBondInAttoeth;
     }
 
-    function calculateValidityBond(uint256 _invalidMarkets, uint256 _totalMarkets, uint256 _targetInvalidMarketsDivisor, uint256 _previousValidityBondInAttoeth) constant public returns (uint256) {
+    function getDesignatedReportStake(IReportingWindow _reportingWindow) public returns (uint256) {
+        uint256 _currentDesignatedReportStakeInAttoRep = designatedReportStakeInAttoRep[_reportingWindow];
+        if (_currentDesignatedReportStakeInAttoRep != 0) {
+            return _currentDesignatedReportStakeInAttoRep;
+        }
+        IReportingWindow _previousReportingWindow = _reportingWindow.getPreviousReportingWindow();
+        uint256 _totalMarketsInPreviousWindow = _reportingWindow.getNumMarkets();
+        uint256 _incorrectDesignatedReportMarketsInPreviousWindow = _reportingWindow.getNumIncorrectDesignatedReportMarkets();
+        uint256 _previousDesignatedReportStakeInAttoRep = designatedReportStakeInAttoRep[_previousReportingWindow];
+
+        _currentDesignatedReportStakeInAttoRep = calculateFloatingValue(_incorrectDesignatedReportMarketsInPreviousWindow, _totalMarketsInPreviousWindow, TARGET_INCORRECT_DESIGNATED_REPORT_MARKETS_DIVISOR, _previousDesignatedReportStakeInAttoRep, Reporting.defaultDesignatedReportStake());
+        designatedReportStakeInAttoRep[_reportingWindow] = _currentDesignatedReportStakeInAttoRep;
+        return _currentDesignatedReportStakeInAttoRep;
+    }
+
+    function calculateFloatingValue(uint256 _invalidMarkets, uint256 _totalMarkets, uint256 _targetDivisor, uint256 _previousValue, uint256 _defaultValue) constant public returns (uint256 _newValue) {
```

Comment:
We may want to rename `invalidMarkets` to something else since `invalid` has very specific meaning in our contracts.  Perhaps `badMarkets`?  (open to suggestions, not a huge fan of `badMarkets`)